### PR TITLE
 use PyIterReturn for gen internal

### PR DESCRIPTION
**Pull request by youknowone**

### CODE REVIEW

File: vm/src/coroutine.rs [hunk 1]
```diff
@@ -1,55 +1,60 @@
...
+            }
         }
     }
 }
 
 #[derive(Debug)]
 pub struct Coro {
     frame: FrameRef,
-    pub closed: AtomicCell<bool>,
+    pub closed: AtomicCell<bool>, // redudant?
```
Comment (DimitrisJim) [inline vm/src/coroutine.rs:31]: If frame was made into an option `closed` could be removed. That's closer to what CPython does.

**Comment by DimitrisJim** [comment]
merged #3185 first so other PRs that might come before this is merged won't get stuck on the failing tests.

File: vm/src/coroutine.rs (hunk 1 above)
Comment (youknowone) [reply vm/src/coroutine.rs]: I replaced this comment to link to this comment

**Comment by youknowone** [comment]
merged by #3202

**Pull request closed without merging**